opcua/server/address_space.py

```diff
@@ -180,34 +180,46 @@ def add_nodes(self, addnodeitems, user=User.Admin):
     def _add_node(self, item, user):
         result = ua.AddNodesResult()
 
-        if item.RequestedNewNodeId in self._aspace:
+        # Check if the null NodeId is given
+        if item.RequestedNewNodeId.is_null():
+            self.logger.debug("RequestedNewNodeId is null")
+            if item.BrowseName.NamespaceIndex:
```

Comment:
I think I found an case where it makes trouble ( in addition to my
subjective aversion to it :-) )
When instantiating a node and subnode, it seems to be common to create the
new node tree with browsename in the same namespace as the original
nodes,while node ID are created in new namespace. Especially this is
required for method arguments otherwise they are broken ( must be in idx
0). But we still want the server to create nodeids, in our custom namespace
and preserve original browsename namespace.

On Thu, Mar 24, 2016, 10:38 Olivier Roulet-Dubonnet <
olivier.roulet@gmail.com> wrote:

> OK merge it. We'll look at it later
>
> On Thu, Mar 24, 2016, 09:11 Denis Štogl <notifications@github.com> wrote:
>
>> In opcua/server/address_space.py
>> <https://github.com/FreeOpcUa/python-opcua/pull/149#discussion_r57284673>
>> :
>>
>> > @@ -180,34 +180,46 @@ def add_nodes(self, addnodeitems, user=User.Admin):
>> >      def _add_node(self, item, user):
>> >          result = ua.AddNodesResult()
>> >
>> > -        if item.RequestedNewNodeId in self._aspace:
>> > +        # Check if the null NodeId is given
>> > +        if item.RequestedNewNodeId.is_null():
>> > +            self.logger.debug("RequestedNewNodeId is null")
>> > +            if item.BrowseName.NamespaceIndex:
>>
>> I it not bad, but still what is a difference if you transport the data in
>> the BrowseName? Both solution are extension of specs :)
>>
>> Sorry that I am stubborn, but I kind a like my solution :)
>>
>> —
>> You are receiving this because you commented.
>>
>>
>> Reply to this email directly or view it on GitHub
>> <https://github.com/FreeOpcUa/python-opcua/pull/149/files/a6be8b95b418d4abec719247991a80265bbcec86#r57284673>
>>
>